app/models/manageiq/providers/openshift/container_manager/container_image.rb

```diff
@@ -1,41 +1,10 @@
-class ManageIQ::Providers::Openshift::ContainerManager::ContainerImage < ContainerImage
-  def annotate_image(annotations)
-    ext_management_system.annotate(
-      "image",
-      digest,
-      annotations
-    )
-  end
-
-  def openscap_summary
-    failed_rules = openscap_rule_results.where(:result => "fail").group(:severity).count
-    [[['High'], 'Critical', 3],
-     [['Medium'], 'Important', 2],
-     [['Low'], 'Medium', 1],
-     [['Info', 'Unknown'], 'Low', 0]].collect do |severities, label, index|
-      {
-        :label         => label,
-        :severityIndex => index,
-        :data          => failed_rules.select { |sev| severities.include?(sev) }.values.sum
-      }
-    end
-  end
+ManageIQ::Providers::Kubernetes::ContainerManager::ContainerImage.include(ActsAsStiLeafClass)
 
-  def security_quality_annotation(compliant)
-    {"quality.images.openshift.io/vulnerability.openscap" => {
-      :name        => "ManageIQ",
-      :timestamp   => Time.now.utc.to_i,
-      :description => "OpenSCAP Score",
-      :reference   => "",
-      :compliant   => compliant,
-      :summary     => openscap_summary
-    }.to_json}
-  end
+class ManageIQ::Providers::Openshift::ContainerManager::ContainerImage < ManageIQ::Providers::Kubernetes::ContainerManager::ContainerImage
+  supports_not :capture
 
-  def annotate_scan_policy_results(causing_policy, compliant)
-    annotate_image({
-      "security.manageiq.org/#{compliant ? "successful" : "failed"}-policy" => causing_policy,
-      "images.openshift.io/deny-execution"  => (!compliant).to_s
-    }.merge!(security_quality_annotation(compliant)))
+  def self.disconnect_inv(ids)
+    _log.info "Disconnecting Images [#{ids}]"
+    base_class.where(:id => ids).update_all(:container_image_registry_id => nil, :deleted_on => Time.now.utc)
```

Comment:
TODO not sure why this is necessary, when the `ContainerImage.disconnect_inv` method is run from the context of the `ManageIQ::Providers::Openshift::ContainerManager::ContainerImage` class `where` isn't picking up on the child classes.  Suspect something to do with the `ActsAsStiLeafClass` but need to investigate more

Reply:
Yes this is due to ActsAsStiLeafClass using just `[sti_class]` for the type clause